Support the AWS_REGION env var

tirsen opened the pull request:
### What problem does this PR solve?

Only override the region if it has been explicitly set, otherwise we clobber the standard AWS_REGION env var

https://github.com/pingcap/dumpling/pull/155#issuecomment-700136706

### What is changed and how it works?

Support the standard AWS_REGION env var


### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Manual test (add detailed scripts or steps below)

kennytm reviewed (commented):
please add a test case to `s3_test.go` (extending `TestApplyUpdate` should be good)

rest LGTM

tirsen commented:
Ah yes I removed the hard coded `us-east-1` and updated the tests. This somewhat breaks backwards compatibility though but probably not in a bad way. Hardcoding `us-east-1` could be disastrous.

kennytm commented:
i think we need a 

```go
if len(aws.StringValue(ses.Config.Region)) == 0 {
    ses.Config.Region = aws.String("please supply --s3.region")
}
```

after line 250 to prevent that `MissingRegion` error thrown by the AWS SDK.
